sunkit_image/enhance.py

```diff
@@ -101,51 +101,42 @@ def mgn(
             "behavior.",
             stacklevel=3,
         )
-
     if weights is None:
         weights = np.ones(len(sigma))
-
     # 1. Replace spurious negative pixels with zero
     if clip is True:
         data[data <= 0] = 1e-15  # Makes sure that all values are above zero
-
+    data = data.astype(np.float32) if data.dtype != np.float32 else data
     image = np.zeros_like(data)
     conv = np.zeros_like(data)
     sigmaw = np.zeros_like(data)
-
     for s, weight in zip(sigma, weights, strict=True):
         # 2 & 3 Create kernel and convolve with image
         # Refer to equation (1) in the paper
-        ndimage.gaussian_filter(data, sigma=s, truncate=truncate, mode="nearest", output=conv)
-
+        conv = ndimage.gaussian_filter(data, sigma=s, truncate=truncate, mode="nearest")
```

Comment:
Why the change here? Presumably the `output` kwarg is used to avoid making a copy of the array.

Reply:
I just don't like it but yeah. I will undo this.